pyro/contrib/gp/util.py

```diff
@@ -66,24 +66,27 @@ def fix_param(self, param, value=None):
             value = getattr(self, param).detach()
         self._fixed_params[param] = value
 
-    def set_mode(self, mode):
+    def set_mode(self, mode, only_this_module=False):
```

Comment:
Maybe rename this to `recursive=True`?

Reply:
Yes, `recursive` seems to be a better name.